spring-ai-client-chat/src/main/java/org/springframework/ai/chat/client/advisor/MessageChatMemoryAdvisor.java

```diff
@@ -109,18 +131,69 @@ private void after(ChatClientResponse chatClientResponse) {
 				.map(g -> (Message) g.getOutput())
 				.toList();
 		}
-		this.getChatMemoryStore().add(this.doGetConversationId(chatClientResponse.context()), assistantMessages);
+		this.chatMemory.add(this.doGetConversationId(chatClientResponse.context()), assistantMessages);
+		return chatClientResponse;
+	}
+
+	public static Builder builder(ChatMemory chatMemory) {
+		return new Builder(chatMemory);
 	}
 
-	public static class Builder extends AbstractChatMemoryAdvisor.AbstractBuilder<ChatMemory> {
+	public static class Builder {
+
+		private String conversationId = ChatMemory.DEFAULT_CONVERSATION_ID;
+
+		private int order = Advisor.DEFAULT_CHAT_MEMORY_PRECEDENCE_ORDER;
+
+		private Scheduler scheduler;
+
+		private ChatMemory chatMemory;
 
 		protected Builder(ChatMemory chatMemory) {
-			super(chatMemory);
+			this.chatMemory = chatMemory;
+		}
+
+		/**
+		 * Set the conversation id.
+		 * @param conversationId the conversation id
+		 * @return the builder
+		 */
+		public Builder conversationId(String conversationId) {
+			this.conversationId = conversationId;
+			return this;
+		}
+
+		/**
+		 * Set whether to protect from blocking.
+		 * @param protectFromBlocking whether to protect from blocking
+		 * @return the builder
+		 */
+		public Builder protectFromBlocking(boolean protectFromBlocking) {
```

Comment:
It can be removed in favour of `scheduler()`

Reply:
see previous comment, it is in other advisors outside chat memory ones.

Reply:
It's only in QuestionAnswerAdvisor for backward compatibility. My suggestion here was based on the fact we are not ensuring backward compatibility, so we could have removed it directly. But that's ok.